lib/Dancer2/Core/Error.pm

```diff
@@ -255,8 +254,13 @@ has content => (
                     }
                 );
             };
+
             # return rendered content unless there was an error.
-            defined $content && return $content;
+            if (defined $content) {
+                # Why is this necessary here and not with other templates e.g. like route templates?
+                utf8::encode($content) if utf8::is_utf8($content); # less confusing via String::UnicodeUTF8: $content = get_utf8($content) if is_unicode($content)
```

Comment:
The stackoverflow thing is a bit alarmist. The use of utf8:: is ok as it is checking the state of the variable. 

If we don't do it the data is garbled, just seemed odd that it need done since it isn't with other template renderings in dancer2.

If the template thing is changed this code will still work right due to the check.

HTH

Sent from my iPhone

> On Feb 13, 2015, at 10:13 AM, Dávid Kovács <notifications@github.com> wrote:
> 
> In lib/Dancer2/Core/Error.pm:
> 
> >              # return rendered content unless there was an error.
> > -            defined $content && return $content;
> > +            if (defined $content) {
> > +                # Why is this necessary here and not with other templates e.g. like route templates?
> > +                utf8::encode($content) if utf8::is_utf8($content); # less confusing via String::UnicodeUTF8: $content = get_utf8($content) if is_unicode($content)
> Is this something to be conserned about? http://stackoverflow.com/a/14584079
> 
> —
> Reply to this email directly or view it on GitHub.
>